urld1e4af3f63_GMIyer_myastro_tgz-pJ8-loader_MatchMakerMainJ8/src/myastro/service/matchmaker/impl/YoniKootaService.java

```diff
@@ -0,0 +1,156 @@
+package myastro.service.matchmaker.impl;
+
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+
+import myastro.constants.Nakshatra;
+import myastro.constants.matchmaker.MatchResult;
+import myastro.service.matchmaker.INakshatraKootaService;
+
+public class YoniKootaService implements INakshatraKootaService {
+
+	private static final YoniKootaService YONI_KOOTA_SERVICE = new YoniKootaService();
+
+	public static INakshatraKootaService getCheckYoniKootaService() {
+		return YONI_KOOTA_SERVICE;
+	}
+
+	private final HashMap<YoniAnimal, YoniAnimal> yoniAnimalEnimityCache;
+	private final HashMap<Nakshatra, NakshatraYoniDetails> nakshatraYoniDetailsCache;
+
+	private YoniKootaService() {
+		yoniAnimalEnimityCache = getYoniAnimalEnimity();
+		nakshatraYoniDetailsCache = getNakshatraYoniDetails();
+	}
+
+	@Override
+	public MatchResult checkKoota(Nakshatra boyStar, Nakshatra girlStar) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 18. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 48-60 into a separate method.